packages/pymedphys_dicom/src/pymedphys_dicom/dicom/_level1/coords.py

```diff
@@ -0,0 +1,195 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License and the additional terms for more
+# details.
+
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+# ADDITIONAL TERMS are also included as allowed by Section 7 of the GNU
+# Affero General Public License. These additional terms are Sections 1, 5,
+# 6, 7, 8, and 9 from the Apache License, Version 2.0 (the "Apache-2.0")
+# where all references to the definition "License" are instead defined to
+# mean the AGPL-3.0+.
+
+# You should have received a copy of the Apache-2.0 along with this
+# program. If not, see <http://www.apache.org/licenses/LICENSE-2.0>.
+
+"""A suite of functions for handling DICOM coordinates"""
+
+import numpy as np
+
+from pymedphys_utilities.libutils import get_imports
+IMPORTS = get_imports(globals())
+
+
+def coords_from_xyz_axes(xyz_axes):
+    """Converts a set of x, y and z axes of a regular grid (e.g. a DICOM
+    pixel array) into an array of three grids whose voxels correspond to
+    and contain the `x`, `y`, and `z` coordinates of the original grid.
+
+    Parameters
+    ----------
+    xyz_axes : tuple
+        A tuple containing three `ndarrays` corresponding to the `x`,
+        `y` and `z` axes of a given 3D grid - usually a DICOM dataset's
+        pixel array.
+
+    Returns
+    -------
+    coords : ndarray
+        An array containing three grids consisting of the `x`, 'y` and
+        `z` coordinates of the corresponding grid (e.g. DICOM dataset's
+        pixel array) from which the original axes were extracted.
+    """
+    ZZ, YY, XX = np.meshgrid(
+        xyz_axes[2], xyz_axes[1], xyz_axes[0], indexing='ij')
+
+    coords = np.array((XX, YY, ZZ), dtype=np.float64)
+    return coords
+
+
+def _orientation_is_head_first(orientation_vector, is_decubitis):
+    if is_decubitis:
+        return np.abs(np.sum(orientation_vector)) != 2
+    else:
+        return np.abs(np.sum(orientation_vector)) == 2
+
+
+def xyz_axes_from_dataset(ds, coord_system="DICOM"):
+    r"""Returns the x, y and z axes of a DICOM dataset's
+    pixel array in the specified coordinate system.
+
+    For DICOM RT Dose datasets, these are the x, y, z axes of the
+    dose grid.
+
+    Parameters
+    ----------
+    ds : pydicom.dataset.Dataset
+        A DICOM dataset that contains pixel data. Supported modalities
+        include 'CT' and 'RTDOSE'.
+
+    coord_system : str, optional
+        The coordinate system in which to return the `x`, `y` and `z`
+        axes of the DICOM dataset. The accepted values of `coord_system`
+        are:
+
+        'DICOM' or 'D':
+            Return axes in the DICOM coordinate system.
+
+        'PATIENT', 'IEC PATIENT' or 'P':
+            Return axes in the IEC patient coordinate system.
+
+        'FIXED', 'IEC FIXED' or 'F':
+            Return axes in the IEC fixed coordinate system.
```

Comment:
Not opposed here, but reasoning for all caps?  Maybe there's value in running an "upper" on user input so that lower case works also? Not sure. Up to you.

Nevermind, I missed it, you did use an upper :).

Reply:
Hehe. Would you prefer lowercase in the docstring?

Reply:
:) I think if they're acronyms they can get away with all caps, such as IEC and DICOM but everything else should probably be lower within the docstring.